kahuna/public/js/search/results.js

```diff
@@ -64,18 +76,38 @@ results.controller('SearchResultsCtrl', [
         // scrollbar becomes hyper-sensitive
         ctrl.maxResults = 5000;
 
+        // If not reloading a previous search, discard any previous
+        // state related to the last search
+        if (! isReloadingPreviousSearch) {
+            lastSearchFirstResultTime = undefined;
+        }
+
         // TODO: avoid this initial search (two API calls to init!)
-        ctrl.searched = search({length: 0}).then(function(images) {
+        ctrl.searched = search({length: 1}).then(function(images) {
```

Comment:
So we can [pull out the uploadTime of the first result](https://github.com/guardian/grid/pull/999/files#diff-b8c6d4770b5b04294cc8c88d27ae65d3R107) in this very first request.